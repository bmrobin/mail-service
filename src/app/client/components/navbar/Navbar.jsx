import React from 'react';
import * as propTypes from 'prop-types';
import { Nav, NavItem } from 'react-bootstrap';

export default class Navbar extends React.Component {

  constructor(props) {
    super(props);
  }

  navSelected(event) {
    this.props.navSelectAction(event);
  }

  render() {
    return (
      <Nav bsStyle="tabs" onSelect={event => this.navSelected(event)}>
        <NavItem eventKey="list" active={this.props.contactListSelected}>Contact List</NavItem>
        <NavItem eventKey="new" active={this.props.newContactSelected}>New Contact</NavItem>
      </Nav>
    );
  }
}

Navbar.propTypes = {
  navSelectAction: propTypes.func,
  contactListSelected: propTypes.bool.isRequired,
  newContactSelected: propTypes.bool.isRequired
};
