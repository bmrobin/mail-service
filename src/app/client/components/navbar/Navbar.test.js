import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Nav } from 'react-bootstrap';
import Navbar from './Navbar';

describe("Navbar Component", () => {

  test("should render component", () => {
    const component = shallow(<Navbar contactListSelected={false} newContactSelected={false} />);
    expect(component.debug()).toMatchSnapshot();
  });

  test("should invoke nav callback when nav selected", () => {
    const mockSelectFn = jest.fn();
    const component = shallow(<Navbar navSelectAction={mockSelectFn} contactListSelected={true} newContactSelected={false} />);
    const arg = {event: 'obj' };
    component.find(Nav).simulate('select', arg);
    expect(mockSelectFn).toHaveBeenCalledWith(arg)
  });

  test("should display contact list when selected", () => {
    const component = renderer.create(<Navbar contactListSelected={true} newContactSelected={false} />).toJSON();
    expect(component).toMatchSnapshot();
  });
  
  test("should display new contact when selected", () => {
    const component = renderer.create(<Navbar contactListSelected={false} newContactSelected={true} />).toJSON();
    expect(component).toMatchSnapshot();
  });

});
