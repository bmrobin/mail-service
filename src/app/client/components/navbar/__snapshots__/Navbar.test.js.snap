// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Component should display contact list when selected 1`] = `
<ul
  className="nav nav-tabs"
  role={null}
>
  <li
    className="active"
    role="presentation"
    style={undefined}
  >
    <a
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      Contact List
    </a>
  </li>
  <li
    className=""
    role="presentation"
    style={undefined}
  >
    <a
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      New Contact
    </a>
  </li>
</ul>
`;

exports[`Navbar Component should display new contact when selected 1`] = `
<ul
  className="nav nav-tabs"
  role={null}
>
  <li
    className=""
    role="presentation"
    style={undefined}
  >
    <a
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      Contact List
    </a>
  </li>
  <li
    className="active"
    role="presentation"
    style={undefined}
  >
    <a
      href="#"
      onClick={[Function]}
      onKeyDown={[Function]}
      role="button"
    >
      New Contact
    </a>
  </li>
</ul>
`;

exports[`Navbar Component should render component 1`] = `
"<Nav bsStyle=\\"tabs\\" onSelect={[Function: onSelect]} justified={false} pullRight={false} pullLeft={false} stacked={false} bsClass=\\"nav\\">
  <NavItem eventKey=\\"list\\" active={false} disabled={false}>
    Contact List
  </NavItem>
  <NavItem eventKey=\\"new\\" active={false} disabled={false}>
    New Contact
  </NavItem>
</Nav>"
`;
